nifi-framework-api/src/main/java/org/apache/nifi/authorization/AbstractPolicyBasedAuthorizer.java

```diff
@@ -337,23 +344,23 @@ public final synchronized AccessPolicy addAccessPolicy(AccessPolicy accessPolicy
      */
     @Override
     public final void checkInheritability(String proposedFingerprint) throws AuthorizationAccessException, UninheritableAuthorizationsException {
+        final PoliciesUsersAndGroups policiesUsersAndGroups;
         try {
             // ensure we understand the proposed fingerprint
-            parsePoliciesUsersAndGroups(proposedFingerprint);
+            policiesUsersAndGroups = parsePoliciesUsersAndGroups(proposedFingerprint);
         } catch (final AuthorizationAccessException e) {
             throw new UninheritableAuthorizationsException("Unable to parse proposed fingerprint: " + e);
         }
 
-        final List<User> users = getSortedUsers();
-        final List<Group> groups = getSortedGroups();
-        final List<AccessPolicy> accessPolicies = getSortedAccessPolicies();
-
-        // ensure we're in a state to inherit
-        if (!users.isEmpty() || !groups.isEmpty() || !accessPolicies.isEmpty()) {
+        if (!isInheritable(policiesUsersAndGroups)) {
```

Comment:
1) `policiesUsersAndGroups` is not used in `isInheritable`.
2) Does it make sense to still have this check if we support force inheriting? Failing on parse makes sense but not sure about this one.

Reply:
Nope.